src/features/users/users.ts

```diff
@@ -0,0 +1,29 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { getUsers } from '../../api/users';
+import { User } from '../../types/User';
+
+type UsersState = {
+  users: User[];
+};
+
+const initialState: UsersState = {
+  users: [],
+};
+
+export const init = createAsyncThunk('users/fetch', () => {
+  return getUsers();
+});
+
+const usersSlice = createSlice({
+  name: 'users',
+  initialState,
+  reducers: {},
+  extraReducers(builder) {
+    builder.addCase(init.fulfilled, (state, action) => {
+      return { ...state, users: action.payload };
+    });
```

Comment:
Consider adding cases for `init.pending` and `init.rejected` in the `extraReducers` to handle loading states and potential errors. This will provide a more complete state management solution by allowing the UI to respond to different stages of the async operation.